devmand/gateway/src/devmand/channels/cli/datastore/DatastoreTransaction.h

```diff
@@ -18,27 +18,45 @@
 #include <ydk/types.hpp>
 #include <atomic>
 
-using std::make_shared;
-using std::map;
-using std::pair;
-using std::shared_ptr;
-using std::string;
-using ydk::Entity;
-using LeafVector = std::vector<pair<string, string>>;
-using ListKeys = std::vector<string>;
 using devmand::channels::cli::datastore::DatastoreDiff;
 using devmand::channels::cli::datastore::DatastoreDiffType;
 using devmand::channels::cli::datastore::DatastoreState;
 using devmand::devices::cli::Model;
 using devmand::devices::cli::ModelRegistry;
 using devmand::devices::cli::Path;
 using folly::dynamic;
+using folly::Optional;
 using folly::parseJson;
 using std::atomic_bool;
+using std::make_shared;
+using std::map;
+using std::multimap;
+using std::pair;
 using std::runtime_error;
+using std::shared_ptr;
+using std::string;
+using std::vector;
+using ydk::Entity;
 
 namespace devmand::channels::cli::datastore {
 
+struct DiffPath {
+  Path path;
+  bool asterix;
+
+  DiffPath() : path("/"), asterix(false) {}
+
+  DiffPath(const Path _path, bool _asterix) : path(_path), asterix(_asterix) {}
+};
+
+struct DiffResult {
+  multimap<Path, DatastoreDiff> diffs;
+  vector<Path> unhandledDiffs;
+  void appendUnhandledPath(Path pathToAppend) {
+    unhandledDiffs.emplace_back(pathToAppend);
+  }
+};
+
 class DatastoreTransaction {
```

Comment:
This needs refactoring:

Splitting functionality (especially the diff)
Hiding utility functions from .h (they make it unreadable)